source/includes/pipeline_config/_000-describe.md.erb

```diff
@@ -0,0 +1,14 @@
+<%=
+describe_object 'The pipeline config object', since: '15.3.0' do
+  label_template            'String',       'The label template to customise the pipeline instance label. Both of material names and ```${COUNT}``` are available in the labeltemplate and the default value of labeltemplate is ```${COUNT}```.'
+  enable_pipeline_locking   'Boolean',      'Whether pipeline is locked to run single instance or not.'
+  name                      'String',       'The name of the pipeline'
+  template                  'String',       'The name of the template used by pipeline.'
+  params                    'Array',        'The list of [parameters](#the-pipeline-parameter-object) to be used across the pipeline or pipeline template.'
+  environment_variables     'Array',        'The list of [environment variables](#the-pipeline-environment-variable-object) defined in the pipeline.'
+  materials                 'Array',        'The list of [materials](#the-pipeline-material-object) to be used by pipeline.'
+  stages                    'Array',        'The list of [stages](#the-stage-object).'
```

Comment:
Would you like to clarify if the `materials` and `stages` array should have atleast one element?